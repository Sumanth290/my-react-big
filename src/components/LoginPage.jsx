import React from "react";
import {connect} from "react-redux";
import {startLogin} from "../actions/auth";

export const LoginPage = (props) => (
    <div className="box-layout">
        {/* <h1>Please login to continue</h1> */}
        <div className="box-layout__fade">
            <div className="box-layout__title">
                <p className="box-layout__heading">Expensify</p>
                <p className="box-layout__text">It's time to get your expenses under control</p>
            </div>
            <div className="box-layout__box">
                <button className="box-layout__button" onClick={props.startLogin}>Login with Google</button>
            </div>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin : () => dispatch(startLogin())
});

export default connect(undefined,mapDispatchToProps)(LoginPage);