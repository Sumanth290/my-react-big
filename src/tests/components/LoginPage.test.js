import React from "react";
import {shallow} from "enzyme";
import {LoginPage} from "../../components/LoginPage.jsx";

test("Should render Login Page correctly",()=>{
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});