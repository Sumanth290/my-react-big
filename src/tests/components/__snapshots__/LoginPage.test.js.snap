// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Login Page correctly 1`] = `
<div>
  <h1>
    Please login to continue
  </h1>
  <button>
    Login
  </button>
</div>
`;
